lib/stream.js

```diff
@@ -83,20 +95,30 @@ const resumeStream = async stream => {
 
 const applyEncoding = (contents, encoding) => encoding === 'buffer' ? new Uint8Array(contents) : contents;
 
+// Handles `childProcess.stdin` aborting or erroring.
+// But ignores the `childProcess.stdin.destroy()` automatically done on child process exit by Node.js.
+// `controller.signal` is used to ensure this never resolves, only rejects.
+const throwOnInputStdioError = ({spawned, stdioStreamsGroups, processExitPromise, controller}) => spawned.stdio
+	.filter((stream, index) => stream !== null && stdioStreamsGroups[index][0].direction === 'input')
+	.map(stream => throwOnInputStreamError(stream, processExitPromise, controller));
+
+const throwOnInputStreamError = async (stream, processExitPromise, {signal}) => {
+	await Promise.all([
+		Promise.race([
+			processExitPromise,
+			finished(stream, {cleanup: true, readable: false, writable: true}),
+		]),
+		once(signal, 'abort'),
```

Comment:
This line is just meant to ensure the function never resolves, since its only purpose is to potentially throw.

Reply:
Maybe add this as a short code comment.

Reply:
Fixed :+1: